tests/tests/subgroup_operations/shader.wgsl

```diff
@@ -0,0 +1,173 @@
+@group(0)
+@binding(0)
+var<storage, read_write> storage_buffer: array<u32>;
+
+@compute
+@workgroup_size(128)
+fn main(
+    @builtin(global_invocation_id) global_id: vec3<u32>,
+    @builtin(num_subgroups) num_subgroups: u32,
+    @builtin(subgroup_id) subgroup_id: u32,
+    @builtin(subgroup_size) subgroup_size: u32,
+    @builtin(subgroup_invocation_id) subgroup_invocation_id: u32,
+) {
+    var passed = 0u;
+    var expected: u32;
+
+    var mask = 1u << 0u;
+    passed |= mask * u32(num_subgroups == 128u / subgroup_size);
```

Comment:
Lets introduce a function that properly does this dance:

```
fn add_result_to_mask(mask: ptr<function, u32>, index: u32, value: bool) {
   (*mask) |= u32(value) << index;
}
```

I think this would clean up this code a bunch